src/banner/banner.service.ts

```diff
@@ -0,0 +1,55 @@
+import { HttpStatus, Injectable } from '@nestjs/common'
+import { InjectModel } from '@nestjs/mongoose'
+import { Model } from 'mongoose'
+import { Banner, BannerDocument } from './schemas/banner.schema'
+import {
+  BannerResponseDto,
+  CreateBannerDto,
+  UpdateBannerDto,
+} from './dto/banner.dto'
+import APIException from 'src/common/dto/APIException.dto'
+
+@Injectable()
+export class BannerService {
+  constructor(
+    @InjectModel(Banner.name) private readonly bannerModel: Model<Banner>,
+  ) {}
+
+  async findAll(): Promise<BannerResponseDto[]> {
+    const docs = await this.bannerModel.find().sort({ endDate: -1 })
+    return docs.map((doc) => this.toResponseDto(doc))
+  }
+
+  async create(dto: CreateBannerDto): Promise<BannerResponseDto> {
+    const created = await this.bannerModel.create(dto)
+    return this.toResponseDto(created)
+  }
+
+  async update(id: string, dto: UpdateBannerDto): Promise<BannerResponseDto> {
+    const updated = await this.bannerModel.findByIdAndUpdate(id, dto, {
+      new: true,
+    })
+
+    if (!updated)
+      throw new APIException(HttpStatus.NOT_FOUND, 'Banner not found.')
```

Comment:
Why not wrapping the if statement?

```suggestion
    if (!updated) {
      throw new APIException(HttpStatus.NOT_FOUND, 'Banner not found.')
    }
```